src/contract_leaderboard/mod.rs

```diff
@@ -0,0 +1,39 @@
+use crate::state::AppState;
+use axum::{extract::State, Json};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ContractEntry {
+    pub address: String,
+    pub base_fees: f64,
+    pub base_fees_usd: Option<f64>,
+}
+
+pub async fn get_contracts(State(state): State<Arc<AppState>>) -> Json<Vec<ContractEntry>> {
```

Comment:
Kinda wild this function takes some app state and returns a Json response ready for sending but 👌 , Rust can be as simple as JS which is one of the things I like about it.

Reply:
"Rust can be as simple as JS " - yes once you grok  lifetimes and learn to leverage the compiler errors definetely. 

Also note that in the `eth-analysis-rs` repo you use `Extension` to manage the shared state where as I use `State` which seems to be the recommended because more typesafe option.

Found this [youtube talk](https://www.youtube.com/watch?v=UKXPleK7GkA&t=267s) on the difference. (only understood about half of it tbh but still found it interesting).

Reply:
No reason, none which I recall anyway. Looks better 👍 .